### PR TITLE
WIP: begin train algorithms

### PR DESCRIPTION
**Pull request by jjjoelle**

### CODE REVIEW

**Comment by jjjoelle** [comment]
https://towardsdatascience.com/how-to-reuse-your-python-models-without-retraining-them-39cd685659a5

for importing files mb?

File: q2_nc/the_end_game.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+tax_col = "taxid_column" 
+taxid = 7742 #vertebraes
+
+tree = TreeClass(taxid, df, tax_col)
+
+for node in tree.get_tree().traverse(strategy="levelorder"):
+    taxid = node.taxid
+    if tree.decision(taxid) is False:
+        continue;
+    for kids in node.get_children():
```
Comment (NathalieFranklin) [inline q2_nc/the_end_game.py:51]: Remove, consider continue instead

File: q2_nc/the_end_game.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+            #here we have enough samples to train tree
+            id_col = "sample_id" 
+            boolIDS = tree.get_identifiers(taxid, id_col)
+            # change create tree to return meta and ft
+            meta, ft = createTree()
+            meta.insert(0, "isChild", boolIDS, True)
+            # change trains to return roc_auc instead of probs
+            roc_auc, estimator = trains (meta, ft) # only estimator is valuable, classifier
+            # can we save in the node?
+            if roc_auc[0] and roc_auc[1] > .5:
```
Comment (NathalieFranklin) [inline q2_nc/the_end_game.py:64]: Lets use the single (True label)

File: q2_nc/the_end_game.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+
+
+
+#CHANGE 
+df = pd.read_csv("metadata.tsv", sep='\t')
+tax_col = "taxid_column" 
+taxid = 7742 #vertebraes
+
+tree = TreeClass(taxid, df, tax_col)
+
```
Comment (NathalieFranklin) [inline q2_nc/the_end_game.py:46]: Change level order to pre- order

**Pull request merged**